cmd/yorkie/version.go

```diff
@@ -17,27 +17,162 @@
 package main
 
 import (
+	"context"
+	"encoding/json"
+	"errors"
 	"fmt"
 	"runtime"
 
+	"connectrpc.com/connect"
 	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"gopkg.in/yaml.v3"
 
+	"github.com/yorkie-team/yorkie/admin"
+	"github.com/yorkie-team/yorkie/api/types"
+	"github.com/yorkie-team/yorkie/cmd/yorkie/config"
 	"github.com/yorkie-team/yorkie/internal/version"
 )
 
+var (
+	clientOnly bool
+	output     string
+)
+
 func newVersionCmd() *cobra.Command {
 	return &cobra.Command{
-		Use:   "version",
-		Short: "Print the version number of Yorkie",
+		Use:     "version",
+		Short:   "Print the version number of Yorkie",
+		PreRunE: config.Preload,
 		RunE: func(cmd *cobra.Command, args []string) error {
-			fmt.Printf("Yorkie: %s\n", version.Version)
-			fmt.Printf("Go: %s\n", runtime.Version())
-			fmt.Printf("Build date: %s\n", version.BuildDate)
+			if err := Validate(); err != nil {
+				return err
+			}
+
+			var versionInfo types.VersionInfo
+			versionInfo.ClientVersion = getYorkieClientVersion()
```

Comment:
How about extracting the code that retrieves server version to the method with the name `getYorkieServerVersion()`, just like client to keep consistency and code readability?